tests/test_particleoperator.py

```diff
@@ -374,3 +374,26 @@ def test_group_works_with_close_coefficients():
     operator += ((1 + 1e-13)) * operator.dagger()
     assert len(operator.group()) == 1
     assert len(operator.group()[0].to_list()) == 2
+
```

Comment:
These tests are good, but need tests for `.modes` and `.swap_with_op_types`. That being said, my 2 cents is that it seems like `.swap_with_op_types` should be a private function instead of public since I can't think of why it would get used by an end-user. IF you agree, then it doesn't _need_ to be rigorously tested since you're not promising its functionality to the end-user.